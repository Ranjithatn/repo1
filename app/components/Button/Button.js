import React, { Fragment } from "react";
import Icon from "../Icon/Icon";

const Button = ({
  leftIcon,
  rightIcon,
  text,
  onClick,
  className,
  disabled = false,
  id,
  type = "button",
  style={},
}) => {
  return (
    <button
      id={id}
      type={type}
      className={"button" + (className !== undefined ? " " + className : "")}
      disabled={disabled}
      onClick={onClick}
      style={ style }
    >
      {leftIcon && <Icon icon={leftIcon} />}
      { text &&<span>{text}</span> }
      {rightIcon && <Icon icon={rightIcon} />}
    </button>
  );
};

export default Button;
