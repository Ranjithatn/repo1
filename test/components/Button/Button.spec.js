import React from "react";
import toJson from "enzyme-to-json";
import Button from "../../../app/components/Button/Button";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";

configure({ adapter: new Adapter() });


describe('<Button />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Button  />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
