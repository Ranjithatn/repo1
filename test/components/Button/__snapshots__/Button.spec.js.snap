// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render without crashing 1`] = `
<Button>
  <button
    className="button"
    disabled={false}
    style={Object {}}
    type="button"
  />
</Button>
`;
